linter/lints/cpcps/lint_crl_has_idp.go

```diff
@@ -153,34 +154,37 @@ func (l *crlHasIDP) Execute(c *x509.RevocationList) *lint.LintResult {
 				Details: "IDP should set onlyContainsUserCerts: TRUE",
 			}
 		}
-	}
-
-	if idpv.PeekASN1Tag(cryptobyte_asn1.Tag(2).ContextSpecific()) {
-		if !idpv.ReadASN1Bytes(&onlyContainsCACerts, cryptobyte_asn1.Tag(2).ContextSpecific()) {
-			return &lint.LintResult{
-				Status:  lint.Error,
-				Details: "Failed to read IDP onlyContainsCACerts",
+	} else {
+		// We read this boolean as a byte and ensure the value is 0xFF because
+		// cryptobyte.ReadASN1Boolean can't handle the custom encoding rules for
+		// the onlyContainsUserCerts[1] and onlyContainsCACerts[2] tagged
+		// fields. X.690 (07/2002) section 8.2 states that a boolean set to true
+		// will have contents FF.
+		// https://www.itu.int/rec/T-REC-X.690-200207-S/en
+		onlyContainsCACerts := make([]byte, 0)
+
+		if idpv.PeekASN1Tag(cryptobyte_asn1.Tag(2).ContextSpecific()) {
+			// If either the onlyContainsCACerts[2] field is present, assume we're
+			// dealing with a CRL containing CA Certs.
```

Comment:
We don't need to do this check -- we're already in the `else` clause of an "if this is a Subscriber CRL", so clearly this must be an CA CRL.